src/client/clientlauncher.cpp

```diff
@@ -59,7 +59,8 @@ ClientLauncher::~ClientLauncher()
 
 	delete g_fontengine;
 
-	device->drop();
+	if (device)
```

Comment:
So, there were 3 commits total on my fork (ordered from newest to oldest)

[deleted] Fixed device->drop()
21b733b6ca5e0706e22afb3811071c73d1286209 Remove SharedPtr, it's not used and will be never used, we use C++11
31cf34ce6af056fe683235d3dce2b54f2d8b34b9 Removed redundant pointer null checks

I used git rebase to consolidate my first commit with the fix, as the change is just one line. This means `git-rebase --interactive` and manually squashing both of my commits together. It, however, created a conflict with 21b733b6ca5e0706e22afb3811071c73d1286209. Resolved it manually by keeping @nerzhul's changes. Now I had just one commit for removing redundant nullchecks and another one for @nerzhul's stuff. But it failed to push to remote as my local copy suddenly didn't have the "Fixed device->drop()" commit. Force pushing essentially allowed me to clean up the mess and act like I didn't do this and fix it in another commit:

```cpp
// if (device) - removed
    device->drop();
```

Instead, it's nice and clean - keeps the repo away from unnecessary commit noise. Hope this clarifies the situation for you.